### PR TITLE
Add and test snow DA increment application code

### PR DESCRIPTION
**Pull request by jiaruidong2017**
Add and test snow DA increment application code to GDASApp, and add a ctest verifying this test works as intended (Issue #200). 

Since there is a bug in the PSL `land-apply_jedi_incr` repo, I used my branch for the test for now. I created the PR for the bugfix, and wait them to merge it. Once the PR is merged, I will switch to use the PSL repo. 

Two datasets are required and listed below:
1) JEDI increment data: 
`/scratch2/NCEPDEV/stmp1/Jiarui.Dong/workdir/jedi/20210323.180000.xainc.sfc_data.tile*.nc`
2) Orography data: 
`/scratch1/NCEPDEV/global/glopara/fix/orog/20220805/C48/C48_oro_data.tile*.nc`

### CODE REVIEW

**Comment by jiaruidong2017** [comment]
@CoryMartin-NOAA Would you please copy the following data into `GDASAPP_TESTDATA`, so that I will modify the script to read these data from `GDASAPP_TESTDATA`? Thanks.

Two datasets are required and listed below: 

JEDI increment data:
`/scratch2/NCEPDEV/stmp1/Jiarui.Dong/workdir/jedi/20210323.180000.xainc.sfc_data.tile*.nc`
Orography data:
`/scratch1/NCEPDEV/global/glopara/fix/orog/20220805/C48/C48_oro_data.tile*.nc`

**Comment by CoryMartin-NOAA** [comment]
@jiaruidong2017 see `$GDASAPP_TESTDATA/land/C48`. Please note that eventually we will be grabbing things like the orography data from the standard GFS/global fix directories, but that won't be until the land DA support is added to the global workflow.

**Comment by jiaruidong2017** [comment]
Thanks @CoryMartin-NOAA for your efforts and comments. I will modify the script to grab data from $GDASAPP_TESTDATA temporally for the ctest.

File: test/land/apply_jedi_incr.sh
Comment (CoryMartin-NOAA) [inline test/land/apply_jedi_incr.sh]: no module purge/load in the test file, the test environment should already be setup properly

File: test/land/apply_jedi_incr.sh
Comment (CoryMartin-NOAA) [inline test/land/apply_jedi_incr.sh:28]: This should have an extra subdirectory under it so that the test is clean (and it should be `rm -rf` before `mkdir -p` to clean it out.

File: test/land/apply_jedi_incr.sh
Comment (CoryMartin-NOAA) [inline test/land/apply_jedi_incr.sh]: as Rahul said before, exit with the return code not a 'made up' one.

**Review by CoryMartin-NOAA — CHANGES_REQUESTED** [review]

File: test/land/apply_jedi_incr.sh
Comment (jiaruidong2017) [reply test/land/apply_jedi_incr.sh]: @CoryMartin-NOAA I see, and I will change it. Thanks.

File: test/land/apply_jedi_incr.sh
Comment (jiaruidong2017) [reply test/land/apply_jedi_incr.sh]: Removed the module load

File: test/land/apply_jedi_incr.sh
Comment (jiaruidong2017) [reply test/land/apply_jedi_incr.sh:28]: Added the `rm -rf $WORKDIR` before `mkdir -p $WORKDIR`.

File: test/land/apply_jedi_incr.sh
Comment (jiaruidong2017) [reply test/land/apply_jedi_incr.sh]: Removed this 'made up' exit code.

File: test/land/apply_jedi_incr.sh
Comment (CoryMartin-NOAA) [inline test/land/apply_jedi_incr.sh]: Probably want a subdirectory and not just 'test/land'

Comment (jiaruidong2017) [reply test/land/apply_jedi_incr.sh]: Okay, I will add` testrun` subdirectory and test it. Thanks.

Comment (CoryMartin-NOAA) [reply test/land/apply_jedi_incr.sh]: I'm thinking more something like `test/land/apply_jedi_incr` so that it is a clean directory for each test.

Comment (jiaruidong2017) [reply test/land/apply_jedi_incr.sh]: Great idea! I will change to `test/land/apply_jediincr` for this ctest. Thanks.

Comment (jiaruidong2017) [reply test/land/apply_jedi_incr.sh]: Added the subdirectory as suggested.

**Review by CoryMartin-NOAA — DISMISSED** [review]
Looks good to me as far as I can tell, will merge once tests pass on Hera and Orion.

**Comment by jiaruidong2017** [comment]
@CoryMartin-NOAA I just modified two previous scripts (create_bkg_ens.sh and test_imsproc.sh) and make them to have same working directories at test/land/subdirectory. I will resubmit the ctests. Thanks.

**Review by CoryMartin-NOAA — APPROVED** [review]

**Comment by CoryMartin-NOAA** [comment]
Looks good thanks @jiaruidong2017

**Pull request merged**
